pgjdbc/src/test/java/org/postgresql/test/TestUtil.java

```diff
@@ -722,4 +726,92 @@ public static void closeQuietly(ResultSet rs) {
       }
     }
   }
+
+  public static void recreateLogicalReplicationSlot(Connection connection, String slotName, String outputPlugin)
+      throws SQLException, InterruptedException {
+    //drop previos slot
+    dropReplicationSlot(connection, slotName);
+
+    PreparedStatement stm = null;
+    try {
+      stm = connection.prepareStatement("SELECT * FROM pg_create_logical_replication_slot(?, ?)");
+      stm.setString(1, slotName);
+      stm.setString(2, outputPlugin);
+      stm.execute();
+    } finally {
+      closeQuietly(stm);
+    }
+  }
+
+  public static void recreatePhysicalReplicationSlot(Connection connection, String slotName)
+      throws SQLException, InterruptedException {
+    //drop previos slot
+    dropReplicationSlot(connection, slotName);
+
+    PreparedStatement stm = null;
+    try {
+      stm = connection.prepareStatement("SELECT * FROM pg_create_physical_replication_slot(?)");
+      stm.setString(1, slotName);
+      stm.execute();
+    } finally {
+      closeQuietly(stm);
+    }
+  }
+
+  public static void dropReplicationSlot(Connection connection, String slotName)
+      throws SQLException, InterruptedException {
+    if (isReplicationSlotActive(connection, slotName)) {
+      if (haveMinimumServerVersion(connection, ServerVersion.v9_5)) {
+        PreparedStatement terminateStatement =
+            connection.prepareStatement(
+                "select pg_terminate_backend(active_pid) from pg_replication_slots "
+                    + "where active = true and slot_name= ?");
+        terminateStatement.setString(1, slotName);
+        terminateStatement.execute();
+        terminateStatement.close();
+      } else {
+        boolean stillActive = true;
+        while (stillActive) {
+          TimeUnit.MILLISECONDS.sleep(100L);
```

Comment:
@Gordiychuk , please add some explicit timeout here. For instance 30 seconds or so. Otherwise Travis builds might get stuck.

Generic rule: every cycle should have explicit timeout.